src/IssueViz/Editor/LocationNavigator.cs

```diff
@@ -47,33 +49,43 @@ internal LocationNavigator(IDocumentNavigator documentNavigator, IIssueSpanCalcu
             this.logger = logger;
         }
 
-        public bool TryNavigate(IAnalysisIssueLocation location)
+        public bool TryNavigate(IAnalysisIssueLocationVisualization locationVisualization)
         {
-            var locationFilePath = location?.FilePath;
-
-            if (string.IsNullOrEmpty(locationFilePath))
+            if (locationVisualization == null)
             {
-                return false;
+                throw new ArgumentNullException(nameof(locationVisualization));
             }
 
             try
             {
-                var textView = documentNavigator.Open(locationFilePath);
-                var locationSpan = spanCalculator.CalculateSpan(location, textView.TextBuffer.CurrentSnapshot);
+                var textView = documentNavigator.Open(locationVisualization.CurrentFilePath);
+                var locationSpan = GetLocationSpan(locationVisualization, textView);
 
-                if (locationSpan.HasValue)
+                if (!locationSpan.IsEmpty)
                 {
-                    documentNavigator.Navigate(textView, locationSpan.Value);
+                    documentNavigator.Navigate(textView, locationSpan);
 
                     return true;
                 }
             }
             catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                logger.WriteLine(Resources.ERR_OpenDocumentException, locationFilePath, ex.Message);
+                logger.WriteLine(Resources.ERR_OpenDocumentException, locationVisualization.CurrentFilePath, ex.Message);
             }
 
             return false;
         }
+
+        private SnapshotSpan GetLocationSpan(IAnalysisIssueLocationVisualization locationVisualization, ITextView textView)
```

Comment:
Weird naming - I wouldn't expect a `GetX` method to set X in some circumstances. I'm not sure what I'd call it, though - `FetchOrCalculateSpan` ?

Reply:
I extracted it outside the method, so now it's
```
var locationSpan = GetLocationSpan(locationVisualization, textView);
locationVisualization.Span = locationSpan;

if (!locationSpan.IsEmpty)
...
```
But I guess it's the side-effect in itself which is problematic -- the fact that TryNavigate is really TryNavigateAndUpdateSpan

Reply:
`RecalculateLocationSpanIfNeeded` ? 🙈